Services/Calendar/classes/class.ilCalendarSchedule.php

```diff
@@ -594,15 +597,44 @@ protected function initPeriod(ilDate $seed)
 				}
 				else
 				{
-					//todo: previous implementation still taking more days than represented in the view.
 					$year_month = $seed->get(IL_CAL_FKT_DATE,'Y-m','UTC');
 					list($year,$month) = explode('-',$year_month);
 
+					#21716
 					$this->start = new ilDate($year_month.'-01',IL_CAL_DATE);
-					$this->start->increment(IL_CAL_DAY,-6);
 
+					$start_unix_time = $this->start->getUnixTime();
+
+					$start_day_of_week = (int)date('w', $start_unix_time);
```

Comment:
An alternative to determine the start_day_of_week is:
$seed->get(IL_CAL_FKT_DATE,'w');

Reply:
I've tested this alternative and it does not fit here. I'm not getting the same result.

Reply:
Ok, thanks.